keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/CheckScheduler.kt

```diff
@@ -61,20 +62,27 @@ class CheckScheduler(
       publisher.publishEvent(ScheduledResourceCheckStarting)
 
       val job = launch {
-        repository
-          .resourcesDueForCheck(resourceCheckMinAgeDuration, resourceCheckBatchSize)
-          .forEach {
-            try {
-              /**
-               * Allow individual resource checks to timeout but catch the `CancellationException`
-               * to prevent the cancellation of all coroutines under [job]
-               */
-              withTimeout(checkTimeout.toMillis()) {
-                launch { resourceActuator.checkResource(it) }
+        runCatching {
+          repository
+            .resourcesDueForCheck(resourceCheckMinAgeDuration, resourceCheckBatchSize)
+        }
+          .onFailure {
+            publisher.publishEvent(ResourceLoadFailed(it))
+          }
+          .onSuccess {
+            it.forEach {
+              try {
+                /**
+                 * Allow individual resource checks to timeout but catch the `CancellationException`
+                 * to prevent the cancellation of all coroutines under [job]
+                 */
+                withTimeout(checkTimeout.toMillis()) {
+                  launch { resourceActuator.checkResource(it) }
+                }
+              } catch (e: TimeoutCancellationException) {
+                log.error("Timed out checking resource ${it.id}", e)
+                publisher.publishEvent(ResourceCheckTimedOut(it.kind, it.id, it.application))
```

Comment:
Does the timeout metric currently appear on our keel dashboard? I don't think I've seen it there.

Reply:
I think it's pretty new and I don't believe it got added to the dashboard, no